patches/0151-lavc-vaapi_dec-Create-VA-parameters-dynamically.patch

```diff
@@ -0,0 +1,104 @@
+From b968b4be4d986506db66609b9130ddcc33e54d16 Mon Sep 17 00:00:00 2001
+From: Fei Wang <fei.w.wang@intel.com>
+Date: Wed, 22 Nov 2023 10:56:55 +0800
+Subject: [PATCH 1/7] lavc/vaapi_dec: Create VA parameters dynamically
+
+Signed-off-by: Fei Wang <fei.w.wang@intel.com>
+---
+ libavcodec/vaapi_decode.c | 29 ++++++++++++++++++++++-------
+ libavcodec/vaapi_decode.h |  7 ++-----
+ 2 files changed, 24 insertions(+), 12 deletions(-)
+
+diff --git a/libavcodec/vaapi_decode.c b/libavcodec/vaapi_decode.c
+index 7c91d50f7b..b52d1377d6 100644
+--- a/libavcodec/vaapi_decode.c
++++ b/libavcodec/vaapi_decode.c
+@@ -39,12 +39,23 @@ int ff_vaapi_decode_make_param_buffer(AVCodecContext *avctx,
+ {
+     VAAPIDecodeContext *ctx = avctx->internal->hwaccel_priv_data;
+     VAStatus vas;
+-    VABufferID buffer;
+ 
+-    av_assert0(pic->nb_param_buffers + 1 <= MAX_PARAM_BUFFERS);
++    av_assert0(pic->nb_param_buffers <= pic->param_allocated);
++    if (pic->nb_param_buffers == pic->param_allocated) {
++        pic->param_buffers =
++            av_realloc_array(pic->param_buffers,
++                             pic->param_allocated + 16,
++                             sizeof(*pic->param_buffers));
++        if (!pic->param_buffers)
```

Comment:
There might be memory leak when pic->param_buffers is not NULL but av_realloc_array returns NULL.